sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationkeyvaultoptions.py

```diff
@@ -16,11 +17,10 @@ class AzureAppConfigurationKeyVaultOptions:
      provided.
     :type secret_clients: list[~azure.keyvault.secrets.SecretClient]
     :param secret_resolver: A function that takes a URI and returns a value.
-    :type secret_resolver: callable
+    :type secret_resolver: Callable[[str], str]
     """
-
-    def __init__(self, credential=None, secret_clients=None, secret_resolver=None):
-        # type: (TokenCredential, List[SecretClient], Callable) -> None
+    def __init__(self, *, credential=None, secret_clients=None, secret_resolver=None):
```

Comment:
Do you want to use something like:

```suggestion
    def __init__(self, *, credential: Optional[TokenCredential]=None, secret_clients: Optional[List[SecretClient]]=None, secret_resolver: Optional[Callable[[str], str]]=None):
```
?

Reply:
This is updated in the other PR as it changes to multiple `@overloads`